commit 280d91b9bff27d96519daa2f14d21e5d402cda2d
Author: Joachim Wiberg <troglobit@gmail.com>
Date:   Mon Dec 6 21:38:20 2021 +0100

    Add support for new libite (-lite) header namespace
    
    Due to an unfortunate name clash with the DirectFB project LiTE, the
    libite (-lite) project had to change its header namespace from
    
       lite/*.h -> libite/*.h
    
    This patch adds support for the new namepace in Finit, triggered by the
    define _LIBITE_LITE, from the .pc file read by pkg-config.  This should
    only be needed on systems that install libite without the compatibility
    symlink lite -> libite/ in the staging include directory.
    
    Signed-off-by: Joachim Wiberg <troglobit@gmail.com>

commit 4ca8256e18656bf547bb4825c0d248761018d121 (HEAD -> master, origin/master, origin/HEAD)
Author: Joachim Wiberg <troglobit@gmail.com>
Date:   Tue Dec 7 00:05:47 2021 +0100

    Follow-up to 280d91b, fix conio.h namespace
    
    Signed-off-by: Joachim Wiberg <troglobit@gmail.com>

commit 393834b253dc5bc9204aa6388c5e484778713fde (HEAD -> master, origin/master, origin/HEAD)
Author: Joachim Wiberg <troglobit@gmail.com>
Date:   Tue Dec 7 00:13:54 2021 +0100

    Follow-up to 4ca8256: ordering mishap
    
    Signed-off-by: Joachim Wiberg <troglobit@gmail.com>


diff --git a/plugins/alsa-utils.c b/plugins/alsa-utils.c
index b465437..9024254 100644
--- a/plugins/alsa-utils.c
+++ b/plugins/alsa-utils.c
@@ -21,7 +21,12 @@
  * THE SOFTWARE.
  */
 
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
+
 #include "finit.h"
 #include "helpers.h"
 #include "plugin.h"
diff --git a/plugins/bootmisc.c b/plugins/bootmisc.c
index 0ab0402..9aa2565 100644
--- a/plugins/bootmisc.c
+++ b/plugins/bootmisc.c
@@ -24,7 +24,11 @@
 #include <ftw.h>
 #include <mntent.h>
 #include <string.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "config.h"
 #include "finit.h"
diff --git a/plugins/dbus.c b/plugins/dbus.c
index c160061..456cc9c 100644
--- a/plugins/dbus.c
+++ b/plugins/dbus.c
@@ -22,7 +22,11 @@
  */
 
 #include <sys/types.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "config.h"
diff --git a/plugins/modprobe.c b/plugins/modprobe.c
index 7cadb78..fdfcb81 100644
--- a/plugins/modprobe.c
+++ b/plugins/modprobe.c
@@ -40,7 +40,11 @@
 #include <sys/stat.h>
 #include <sys/time.h>		/* gettimeofday() */
 #include <sys/types.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "config.h"
 #include "finit.h"
diff --git a/plugins/modules-load.c b/plugins/modules-load.c
index de4f892..7781ca6 100644
--- a/plugins/modules-load.c
+++ b/plugins/modules-load.c
@@ -21,10 +21,14 @@
  * THE SOFTWARE.
  */
 
-#include <lite/lite.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <dirent.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "helpers.h"
diff --git a/plugins/procps.c b/plugins/procps.c
index 8f38220..117cbb5 100644
--- a/plugins/procps.c
+++ b/plugins/procps.c
@@ -22,7 +22,11 @@
  */
 
 #include <glob.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "helpers.h"
diff --git a/plugins/resolvconf.c b/plugins/resolvconf.c
index 9b1d9b5..aae13db 100644
--- a/plugins/resolvconf.c
+++ b/plugins/resolvconf.c
@@ -21,7 +21,11 @@
  * THE SOFTWARE.
  */
 
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "helpers.h"
diff --git a/plugins/rtc.c b/plugins/rtc.c
index b77f583..90a6832 100644
--- a/plugins/rtc.c
+++ b/plugins/rtc.c
@@ -24,7 +24,11 @@
 #include <time.h>
 #include <sys/ioctl.h>
 #include <linux/rtc.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "helpers.h"
diff --git a/plugins/tty.c b/plugins/tty.c
index 6b98856..1d2bdce 100644
--- a/plugins/tty.c
+++ b/plugins/tty.c
@@ -27,7 +27,11 @@
 #include <unistd.h>		/* read() */
 #include <sys/inotify.h>
 #include <sys/stat.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "helpers.h"
diff --git a/plugins/urandom.c b/plugins/urandom.c
index 59d7763..67cf3ee 100644
--- a/plugins/urandom.c
+++ b/plugins/urandom.c
@@ -25,7 +25,11 @@
 #include <sys/stat.h>
 #include <sys/time.h>		/* gettimeofday() */
 #include <sys/types.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "config.h"
 #include "finit.h"
diff --git a/plugins/x11-common.c b/plugins/x11-common.c
index 84025f9..77a8c28 100644
--- a/plugins/x11-common.c
+++ b/plugins/x11-common.c
@@ -24,7 +24,11 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "helpers.h"
diff --git a/src/api.c b/src/api.c
index f9276f6..7a0198f 100644
--- a/src/api.c
+++ b/src/api.c
@@ -29,7 +29,11 @@
 #include <unistd.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 #include <uev/uev.h>
 
 #include "config.h"
diff --git a/src/cgroup.c b/src/cgroup.c
index ac9ad4e..a15bf93 100644
--- a/src/cgroup.c
+++ b/src/cgroup.c
@@ -24,8 +24,13 @@
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
-#include <lite/lite.h>
-#include <lite/queue.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+# include <libite/queue.h>	/* BSD sys/queue.h API */
+#else
+# include <lite/lite.h>
+# include <lite/queue.h>	/* BSD sys/queue.h API */
+#endif
 #include <sys/mount.h>
 #include <sys/sysinfo.h>		/* get_nprocs_conf() */
 
diff --git a/src/cgutil.c b/src/cgutil.c
index 1cdf6f5..f294ec8 100644
--- a/src/cgutil.c
+++ b/src/cgutil.c
@@ -30,7 +30,11 @@
 #include <stdlib.h>
 #include <search.h>
 #include <inttypes.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 #include <sys/sysinfo.h>		/* sysinfo() */
 
 #include "cgutil.h"
diff --git a/src/cond-w.c b/src/cond-w.c
index c482aab..8ed0f98 100644
--- a/src/cond-w.c
+++ b/src/cond-w.c
@@ -23,8 +23,12 @@
 
 #include <ftw.h>
 #include <libgen.h>
-#include <lite/lite.h>
 #include <stdio.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "cond.h"
diff --git a/src/cond.c b/src/cond.c
index cf8d3d7..5ed6478 100644
--- a/src/cond.c
+++ b/src/cond.c
@@ -21,8 +21,12 @@
  * THE SOFTWARE.
  */
 
-#include <lite/lite.h>
 #include <stdio.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "cond.h"
diff --git a/src/conf.c b/src/conf.c
index 961516f..13833d0 100644
--- a/src/conf.c
+++ b/src/conf.c
@@ -28,8 +28,13 @@
 #include <string.h>
 #include <sys/inotify.h>
 #include <sys/resource.h>
-#include <lite/lite.h>
-#include <lite/queue.h>		/* BSD sys/queue.h API */
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+# include <libite/queue.h>	/* BSD sys/queue.h API */
+#else
+# include <lite/lite.h>
+# include <lite/queue.h>	/* BSD sys/queue.h API */
+#endif
 #include <time.h>
 #include <glob.h>
 
diff --git a/src/exec.c b/src/exec.c
index f1876e1..00cb743 100644
--- a/src/exec.c
+++ b/src/exec.c
@@ -33,7 +33,11 @@
 #include <sys/wait.h>
 #include <sys/ttydefaults.h>	/* Not included by default in musl libc */
 #include <termios.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 #include <linux/vt.h>
 
 #include "finit.h"
diff --git a/src/finit.c b/src/finit.c
index 58dad97..5bf659e 100644
--- a/src/finit.c
+++ b/src/finit.c
@@ -36,7 +36,11 @@
 #include <sys/mount.h>
 #include <sys/stat.h>		/* umask(), mkdir() */
 #include <sys/wait.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "cgroup.h"
diff --git a/src/finit.h b/src/finit.h
index 562dbf1..e2a4d62 100644
--- a/src/finit.h
+++ b/src/finit.h
@@ -34,7 +34,11 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 #include <uev/uev.h>
 
 /* just in case */
diff --git a/src/helpers.c b/src/helpers.c
index 64f4138..845cb2a 100644
--- a/src/helpers.c
+++ b/src/helpers.c
@@ -31,7 +31,11 @@
 #include <netinet/in.h>
 #include <stdarg.h>
 #include <sys/ioctl.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "helpers.h"
diff --git a/src/helpers.h b/src/helpers.h
index 1ca9ff9..8384046 100644
--- a/src/helpers.h
+++ b/src/helpers.h
@@ -32,7 +32,11 @@
 #include <sys/types.h>
 #include <sys/ttydefaults.h>	/* Not included by default in musl libc */
 #include <termios.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 #include "log.h"
 
 #define PROGRESS_DEFAULT PROGRESS_MODERN
diff --git a/src/initctl.c b/src/initctl.c
index feb0865..8e07325 100644
--- a/src/initctl.c
+++ b/src/initctl.c
@@ -33,7 +33,11 @@
 #include <time.h>
 #include <utmp.h>
 #include <arpa/inet.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "client.h"
 #include "cond.h"
diff --git a/src/iwatch.h b/src/iwatch.h
index adff92a..decf960 100644
--- a/src/iwatch.h
+++ b/src/iwatch.h
@@ -26,7 +26,11 @@
 #define FINIT_IWATCH_H_
 
 #include <stdint.h>
-#include <lite/queue.h>
+#ifdef _LIBITE_LITE
+# include <libite/queue.h>	/* BSD sys/queue.h API */
+#else
+# include <lite/queue.h>	/* BSD sys/queue.h API */
+#endif
 #include <sys/inotify.h>
 
 #ifndef IN_MASK_CREATE
diff --git a/src/keventd.c b/src/keventd.c
index 0794f75..71c7de2 100644
--- a/src/keventd.c
+++ b/src/keventd.c
@@ -37,7 +37,11 @@
 #include <linux/types.h>
 #include <linux/netlink.h>
 
-#include "lite/lite.h"
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "cond.h"
 #include "pid.h"
diff --git a/src/log.c b/src/log.c
index 33fa63f..4c9a48b 100644
--- a/src/log.c
+++ b/src/log.c
@@ -24,7 +24,11 @@
 #include "config.h"
 #include <stdio.h>
 #include <stdarg.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "helpers.h"
diff --git a/src/logit.c b/src/logit.c
index 7550e44..49de16e 100644
--- a/src/logit.c
+++ b/src/logit.c
@@ -30,7 +30,11 @@
 #include <syslog.h>
 #include <unistd.h>
 #include <sys/stat.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 static const char version_info[] = PACKAGE_NAME " v" PACKAGE_VERSION;
 extern int logrotate(char *file, int num, off_t sz);
diff --git a/src/logrotate.c b/src/logrotate.c
index 454447b..c2707ce 100644
--- a/src/logrotate.c
+++ b/src/logrotate.c
@@ -20,12 +20,16 @@
  * THE SOFTWARE.
  */
 
-#include <lite/lite.h>
 #include <stdio.h>
 #include <syslog.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 static int recreate(char *path, mode_t mode, uid_t uid, gid_t gid)
 {
diff --git a/src/pid.c b/src/pid.c
index 809cc42..926c016 100644
--- a/src/pid.c
+++ b/src/pid.c
@@ -27,7 +27,11 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "pid.h"
 #include "svc.h"
diff --git a/src/plugin.c b/src/plugin.c
index b005d09..44bea5d 100644
--- a/src/plugin.c
+++ b/src/plugin.c
@@ -26,8 +26,13 @@
 #include <dirent.h>		/* readdir() et al */
 #include <poll.h>
 #include <string.h>
-#include <lite/lite.h>
-#include <lite/queue.h>		/* BSD sys/queue.h API */
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+# include <libite/queue.h>	/* BSD sys/queue.h API */
+#else
+# include <lite/lite.h>
+# include <lite/queue.h>	/* BSD sys/queue.h API */
+#endif
 
 #include "config.h"
 #include "cond.h"
diff --git a/src/plugin.h b/src/plugin.h
index 0727aa1..83d9891 100644
--- a/src/plugin.h
+++ b/src/plugin.h
@@ -25,7 +25,11 @@
 #define FINIT_PLUGIN_H_
 
 #include <uev/uev.h>
-#include <lite/queue.h>		/* BSD sys/queue.h API */
+#ifdef _LIBITE_LITE
+# include <libite/queue.h>	/* BSD sys/queue.h API */
+#else
+# include <lite/queue.h>	/* BSD sys/queue.h API */
+#endif
 
 #include "svc.h"
 
diff --git a/src/serv.c b/src/serv.c
index 7b5e6fe..b5621ad 100644
--- a/src/serv.c
+++ b/src/serv.c
@@ -28,7 +28,11 @@
 #include <glob.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "initctl.h"
 
diff --git a/src/service.c b/src/service.c
index 73905c8..c0d646f 100644
--- a/src/service.c
+++ b/src/service.c
@@ -31,7 +31,11 @@
 #include <sys/resource.h>
 #include <sys/wait.h>
 #include <net/if.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 #include <wordexp.h>
 
 #include "cgroup.h"
diff --git a/src/sig.c b/src/sig.c
index b0e44aa..43b0d80 100644
--- a/src/sig.c
+++ b/src/sig.c
@@ -73,7 +73,11 @@
 #include <string.h>		/* strerror() */
 #include <sys/reboot.h>
 #include <sys/wait.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "finit.h"
 #include "cond.h"
diff --git a/src/svc.c b/src/svc.c
index 835321b..ec26a9c 100644
--- a/src/svc.c
+++ b/src/svc.c
@@ -29,7 +29,11 @@
 #include <stdlib.h>
 #include <strings.h>
 #include <sys/time.h>
-#include <lite/queue.h>		/* BSD sys/queue.h API */
+#ifdef _LIBITE_LITE
+# include <libite/queue.h>	/* BSD sys/queue.h API */
+#else
+# include <lite/queue.h>	/* BSD sys/queue.h API */
+#endif
 
 #include "finit.h"
 #include "svc.h"
diff --git a/src/svc.h b/src/svc.h
index da375d5..a0ff78c 100644
--- a/src/svc.h
+++ b/src/svc.h
@@ -28,8 +28,13 @@
 #include <sys/ipc.h>		/* IPC_CREAT */
 #include <sys/resource.h>
 #include <sys/types.h>		/* pid_t */
-#include <lite/lite.h>
-#include <lite/queue.h>		/* BSD sys/queue.h API */
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+# include <libite/queue.h>	/* BSD sys/queue.h API */
+#else
+# include <lite/lite.h>
+# include <lite/queue.h>	/* BSD sys/queue.h API */
+#endif
 #include <uev/uev.h>
 
 #include "cgroup.h"
diff --git a/src/tty.c b/src/tty.c
index aea4af7..49602fb 100644
--- a/src/tty.c
+++ b/src/tty.c
@@ -29,7 +29,11 @@
 #include <sys/prctl.h>
 #include <sys/types.h>
 #include <sys/wait.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "config.h"		/* Generated by configure script */
 #include "finit.h"
diff --git a/src/util.c b/src/util.c
index 959581b..28f4c92 100644
--- a/src/util.c
+++ b/src/util.c
@@ -37,7 +37,11 @@
 # include <sys/ioctl.h>
 #endif
 #include <sys/sysinfo.h>	/* sysinfo() */
-#include <lite/lite.h>		/* strlcat() */
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 #include "util.h"
 
 #ifdef HAVE_TERMIOS_H
diff --git a/src/util.h b/src/util.h
index 45c6654..025a0c6 100644
--- a/src/util.h
+++ b/src/util.h
@@ -32,7 +32,6 @@
 #include <sys/reboot.h>
 #include <termios.h>
 #endif
-#include <lite/lite.h>
 
 #ifndef RB_SW_SUSPEND /* Since Linux 2.5.18, but not in GLIBC until 2.19 */
 #define RB_SW_SUSPEND	0xd000fce2
@@ -45,7 +44,13 @@ extern char *prognm;
 
 /* Define for printheader() from conio.h, used in initctl */
 #define SCREEN_WIDTH ttcols
-#include <lite/conio.h>
+#ifdef _LIBITE_LITE
+# include <libite/conio.h>
+# include <libite/lite.h>
+#else
+# include <lite/conio.h>
+# include <lite/lite.h>
+#endif

 char *progname     (char *arg0);
 
diff --git a/src/utmp-api.c b/src/utmp-api.c
index 8b0ba19..a16fe1f 100644
--- a/src/utmp-api.c
+++ b/src/utmp-api.c
@@ -26,7 +26,11 @@
 #include <time.h>
 #include <arpa/inet.h>
 #include <sys/utsname.h>
-#include <lite/lite.h>
+#ifdef _LIBITE_LITE
+# include <libite/lite.h>
+#else
+# include <lite/lite.h>
+#endif
 
 #include "helpers.h"
 #include "util.h"
